src/ws/session.c

```diff
@@ -620,6 +624,17 @@ session (char **env)
 
   fd_remap (remap_fds, n_remap_fds);
 
+  /* close all other fds, mostly from PAM */
+  for (int fd = n_remap_fds; fd <= SESSION_MAX_FDS; ++fd)
+   close (fd);
+  /* sanity check: the last one should really not be open */
+  if (errno != EBADF)
```

Comment:
Need to set errno to 0 before the last close and test it separately.

Reply:
Alternatively:

```
int last_errno = 0;
for (int fd = n_remap_fds; fd <= SESSION_MAX_FDS; ++fd)
   if (close (fd) < 0)
     last_errno = errno;
```